### PR TITLE
Upgrade Flask MV to 3.0

### LINKED ISSUE #10067: Support Flask 3.0 in MLflow
### Summary

We currently pin `Flask` to `< 3.0`. Flask 3.0 was recently released. We should update the requirement:

```diff
diff --git a/requirements/core-requirements.txt b/requirements/core-requirements.txt
index 99f4cdd62..11f7e3770 100644
--- a/requirements/core-requirements.txt
+++ b/requirements/core-requirements.txt
@@ -4,7 +4,7 @@
 
 alembic<2,!=1.10.0
 docker<7,>=4.0.0
-Flask<3
+Flask<4
 numpy<2
 scipy<2
 pandas<3
diff --git a/requirements/core-requirements.yaml b/requirements/core-requirements.yaml
index 5447b07c7..4ffb44c6b 100644
--- a/requirements/core-requirements.yaml
+++ b/requirements/core-requirements.yaml
@@ -17,7 +17,7 @@ docker:
 
 flask:
   pip_release: Flask
-  max_major_version: 2
+  max_major_version: 3
 
 numpy:
   pip_release: numpy
```

References:

- https://pypi.org/project/Flask/3.0.0/

### Notes

- Make sure to open a PR from a **non-master** branch.
- Sign off the commit using the `-s` flag when making a commit:

  ```sh
  git commit -s -m "..."
           # ^^ make sure to use this
  ```

- Include `#{issue_number}` (e.g. `#123`) in the PR description when opening a PR.

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/10098?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10098/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10098
```

</p>
</details>

### Related Issues/PRs

Resolve #10067 

### What changes are proposed in this pull request?

Bump up Flask major version from 2 to 3.

Supposedly this version update doesn't do much ([ref](https://flask.palletsprojects.com/en/3.0.x/changes/)), and only required code change on our end is to remove usage of `__version__` attribute, which was used [here](https://github.com/mlflow/mlflow/blob/299c229e874b68fcb65928b2e690cb31a10dc241/mlflow/server/__init__.py#L43).

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [x] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [x] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
@B-Step62 Do you see the auto-merge button?

image

**Comment by B-Step62** [comment]
Thx for the quick review @harupy!

> @B-Step62 Do you see the auto-merge button?

Yup, what is the common practice here, can I turn it on?

File: .github/workflows/requirements.yml [hunk 1]
```diff
@@ -79,4 +79,5 @@ jobs:
       - name: Run tests
         run: |
           source dev/setup-ssh.sh
-          ./dev/run-python-tests.sh
+          pytest tests --quiet --requires-ssh --ignore-flavors \
```
Comment (B-Step62) [inline .github/workflows/requirements.yml:82]: ref: https://github.com/mlflow/mlflow/commit/54d463091fa12b641dec26f59d3bbdba78e059ac

**Comment by harupy** [comment]
It's up to you :) Manual merge is also fine :)

**Comment by harupy** [comment]
@B-Step62 The following change should be able to fix the error:

```diff
diff --git a/.github/workflows/requirements.yml b/.github/workflows/requirements.yml
index 679145e25..85c620eea 100644
--- a/.github/workflows/requirements.yml
+++ b/.github/workflows/requirements.yml
@@ -18,6 +18,7 @@ concurrency:
   cancel-in-progress: true
 
 env:
+  MLFLOW_HOME: /home/runner/work/mlflow/mlflow
   MLFLOW_CONDA_HOME: /usr/share/miniconda
   SPARK_LOCAL_IP: localhost
   PYTHON_VERSION: "3.8"
```

**Comment by harupy** [comment]
We can merge this PR and make the change above as a follow-up.

**Comment by B-Step62** [comment]
Thanks, merging CR and will create a quick PR for the follow-up

**Pull request merged**